### ISSUE TITLE
Initialize failed with lastest code 267379d

### ISSUE THREAD

**Issue by yorkane** [if-ai/ComfyUI-IF_AI_tools#134]
Cannot import /home/aigc/ComfyUI/custom_nodes/ComfyUI-IF_AI_tools module for custom nodes: cannot import name 'save_combo_settings' from 'ComfyUI-IF_AI_tools.utils' (/home/aigc/ComfyUI/custom_nodes/ComfyUI-IF_AI_tools/utils.py)

267379d

**Comment by liubin79813** [comment]
same problem

**Comment by wolfden** [comment]
ditto

ImportError: cannot import name 'save_combo_settings' from 'ComfyUI-IF_AI_tools.utils' (F:\ComfyUI\custom_nodes\ComfyUI-IF_AI_tools\utils.py)

Cannot import F:\ComfyUI\custom_nodes\ComfyUI-IF_AI_tools module for custom nodes: cannot import name 'save_combo_settings' from 'ComfyUI-IF_AI_tools.utils' (F:\ComfyUI\custom_nodes\ComfyUI-IF_AI_tools\utils.py)

**Comment by PyrateGFXProductions** [comment]
So, it's not just me...
I spent over 4 hrs last night trying to fix this error with no luck...

**Comment by jakechai** [comment]
missing utils:
![missing utils](https://github.com/user-attachments/assets/aeb8968b-d1e4-4c40-8ee9-d1f67fbb8d0f)

**Comment by antique-goo** [comment]
same error

**Comment by jacques42** [comment]
Same here I am sorry :(

**Comment by mkat2** [comment]
same error...is there a solution? all happened after updating the Comfyui-if-ai_tools to the latest version.... the new file also doesnt work: https://github.com/if-ai/ComfyUI-IF_AI_tools/commit/267379d8539f596166a5b7f6882eba3a42e34c4b

only get the node working by removing those three last save/load/create items from IFImagePromptNode.py... but is ther a good fix that can use those items??

**Comment by PyrateGFXProductions** [comment]
adding the following lines to utils.py fixed it for me...

`def save_combo_settings(settings: Dict[str, Any], file_path: str) -> None:
    """
    Save the combo settings to a YAML file.
    
    Args:
        settings (dict): The settings to save.
        file_path (str): The path to the file where settings will be saved.
    """
    try:
        dump_yaml(settings, file_path)  # Use the existing dump_yaml function
    except Exception as e:
        logger.error(f"Error saving settings to {file_path}: {e}")

def load_combo_settings(file_path: str) -> Dict[str, Any]:
    """
    Load the combo settings from a YAML file.
    
    Args:
        file_path (str): The path to the file from which settings will be loaded.
    
    Returns:
        dict: The loaded settings.
    """
    try:
        with open(file_path, 'r') as yaml_file:
            settings = yaml.safe_load(yaml_file)
        return settings
    except FileNotFoundError:
        logger.warning(f"Settings file not found: {file_path}")
        return {}  # Return an empty dict if the file is not found
    except yaml.YAMLError as e:
        logger.error(f"Error loading YAML from the settings file: {file_path} - {e}")
        return {}

def create_settings_from_ui_IFImagePromptNode(ui_data: Dict[str, Any]) -> Dict[str, Any]:
    """
    Create settings from the UI data for the IFImagePromptNode.
    
    Args:
        ui_data (dict): The data from the UI.
    
    Returns:
        dict: The created settings.
    """
    settings = {
        'image_size': ui_data.get('image_size', (512, 512)),  # Default to 512x512
        'model': ui_data.get('model', 'default_model'),  # Default model name
        'use_mask': ui_data.get('use_mask', False),  # Default to not using a mask
        # Add more parameters as needed based on your UI
    }
    return settings`

**Comment by PyrateGFXProductions** [comment]
adding the following code to the end of utils.py fixed it for me.

```
def save_combo_settings(settings: Dict[str, Any], file_path: str) -> None:
    """
    Save the combo settings to a YAML file.
    
    Args:
        settings (dict): The settings to save.
        file_path (str): The path to the file where settings will be saved.
    """
    try:
        dump_yaml(settings, file_path)  # Use the existing dump_yaml function
    except Exception as e:
        logger.error(f"Error saving settings to {file_path}: {e}")

def load_combo_settings(file_path: str) -> Dict[str, Any]:
    """
    Load the combo settings from a YAML file.
    
    Args:
        file_path (str): The path to the file from which settings will be loaded.
    
    Returns:
        dict: The loaded settings.
    """
    try:
        with open(file_path, 'r') as yaml_file:
            settings = yaml.safe_load(yaml_file)
        return settings
    except FileNotFoundError:
        logger.warning(f"Settings file not found: {file_path}")
        return {}  # Return an empty dict if the file is not found
    except yaml.YAMLError as e:
        logger.error(f"Error loading YAML from the settings file: {file_path} - {e}")
        return {}

def create_settings_from_ui_IFImagePromptNode(ui_data: Dict[str, Any]) -> Dict[str, Any]:
    """
    Create settings from the UI data for the IFImagePromptNode.
    
    Args:
        ui_data (dict): The data from the UI.
    
    Returns:
        dict: The created settings.
    """
    settings = {
        'image_size': ui_data.get('image_size', (512, 512)),  # Default to 512x512
        'model': ui_data.get('model', 'default_model'),  # Default model name
        'use_mask': ui_data.get('use_mask', False),  # Default to not using a mask
        # Add more parameters as needed based on your UI
    }
    return settings
```

**Comment by mkat2** [comment]
> adding the following code to the end of utils.py fixed it for me.
> 
> ```
> def save_combo_settings(settings: Dict[str, Any], file_path: str) -> None:
>     """
>     Save the combo settings to a YAML file.
>     
>     Args:
>         settings (dict): The settings to save.
>         file_path (str): The path to the file where settings will be saved.
>     """
>     try:
>         dump_yaml(settings, file_path)  # Use the existing dump_yaml function
>     except Exception as e:
>         logger.error(f"Error saving settings to {file_path}: {e}")
> 
> def load_combo_settings(file_path: str) -> Dict[str, Any]:
>     """
>     Load the combo settings from a YAML file.
>     
>     Args:
>         file_path (str): The path to the file from which settings will be loaded.
>     
>     Returns:
>         dict: The loaded settings.
>     """
>     try:
>         with open(file_path, 'r') as yaml_file:
>             settings = yaml.safe_load(yaml_file)
>         return settings
>     except FileNotFoundError:
>         logger.warning(f"Settings file not found: {file_path}")
>         return {}  # Return an empty dict if the file is not found
>     except yaml.YAMLError as e:
>         logger.error(f"Error loading YAML from the settings file: {file_path} - {e}")
>         return {}
> 
> def create_settings_from_ui_IFImagePromptNode(ui_data: Dict[str, Any]) -> Dict[str, Any]:
>     """
>     Create settings from the UI data for the IFImagePromptNode.
>     
>     Args:
>         ui_data (dict): The data from the UI.
>     
>     Returns:
>         dict: The created settings.
>     """
>     settings = {
>         'image_size': ui_data.get('image_size', (512, 512)),  # Default to 512x512
>         'model': ui_data.get('model', 'default_model'),  # Default model name
>         'use_mask': ui_data.get('use_mask', False),  # Default to not using a mask
>         # Add more parameters as needed based on your UI
>     }
>     return settings
> ```

Yes many thanks, adding these defs at least makes the node load correctly!

**Comment by wolfden** [comment]
I take this mod is dead, always busted

**Comment by PyrateGFXProductions** [comment]
> I take this mod is dead, always busted

I don't think so. The author is active and has an active Youtube channel but likely busy with other projects as he hasn't touched this in 2 weeks.
